clean-leather.lic

```diff
@@ -2,80 +2,156 @@
-    DRC.message("@preservative is set to #{@preservative}.") if args.debug
-    @item = args.noun =~ /^bones?$/i ? '7' : '8'
-    DRC.message("@item is set to #{@item}.") if args.debug
-    @speed = args.speed || ''
-
-    while bput("get #{args.noun} from my #{args.source}", 'You get', 'You carefully remove', 'What were you') != 'What were you'
-      fput('stow left') if left_hand == 'bundling rope'
-      fput('stow right') if right_hand == 'bundling rope'
-      get_crafting_item('scraper', @bag, @bag_items, @belt)
-      until bput("scrape #{args.noun} with my scraper #{@speed}", 'roundtime', 'looks as clean as you') == 'looks as clean as you'
-        waitrt?
-      end
-      waitrt?
-      stow_crafting_item('scraper', @bag, @belt)
-      if bput("get my #{@preservative}", 'You get', 'What were you') == 'What were you'
-        order_stow_lotion
-        DRCT.walk_to(@room)
+    # Get as many skins as we have in the source container, cleaning them, then preserving
+    # them, then putting them in the storage container.
+    loop do
+      case DRC.bput("get #{animal} #{hide} from my #{source}", /^You get/i, /^You carefully remove/i, /^What were you/i, /What do you want to get?/i)
+      when /^You get/i, /^You carefully remove/i
+        scrape_leather(hide, speed)
+        apply_preservative(animal, hide, storage, full_preservative)
+      when /^What were you/i, /What do you want to get?/i
+        DRC.message("Done with all #{animal} #{hide}s.")
+        break
+      else
+        DRC.message("Unable to get the rawhide. Please report this error.")
+        DRCI.stow_hands
       end
-      bput("pour #{@preservative} on my #{args.noun}", 'roundtime')
-      waitrt?
+    end
+  end
 
-      if @bag.nil?
-        fput("stow my #{@preservative}")
+  def scrape_leather(hide, speed)
+    DRCC.get_crafting_item('scraper', @bag, @bag_items, @belt)
+
+    # Scrapes a skin until it's fully-scraped
+    loop do
+      case DRC.bput("scrape #{hide} with my scraper #{speed}", /^You (quickly|carefully)? scrape your/i, /looks as clean as you/i)
+      when /^You (quickly|carefully)? scrape your/i
+        waitrt?
+        next
+      when /looks as clean as you/
+        break
       else
-        fput("put my #{@preservative} in my #{@bag}")
+        DRC.message("Unable to scrape leather. Please report this error.")
+        DRCI.stow_hands
+        exit
+      end
+    end
+
+    DRCC.stow_crafting_item('scraper', @bag, @belt)
+  end
+
+  def apply_preservative(animal, hide, storage, full_preservative)
+
+    # This loop applies a single application of preservative unless
+    # the user has specified "full" application in the args, in which case
+    # it applies until game messages suggest we're at max.
+    loop do
+      # If we have no preservative on-hand, get money,
+      # get preservative, go back to crafting spot.
+      unless DRCI.get_item_if_not_held?(@preservative)
+        DRCM.ensure_copper_on_hand(2000, @settings)
```

Comment:
`common-money` needs added to `custom_require`

Reply:
Great catch. Thanks.